File: Workspace/week10/JSW/기본 문제/1629/solution_01.cpp
#include <iostream>

using namespace std;
int main() {
    int A, B, C;
    long long result = 1;
    cin >> A >> B >> C;
    for (int i = 0; i < B; i++) {
        result *= A;
        result %= C;
    }
    cout << result << endl;
    return 0;
}

File: Workspace/week10/JSW/기본 문제/1629/solution_01.py
# Problem: 곱셈(https://www.acmicpc.net/problem/1629)
# Date: 2024.12.05
# Comment: Naive Approach -> failed
    
A, B, C = map(int, input().split())
A %= C
result = 1
for _ in range(B):
    result *= A
    result %= C
    print(result)
print(result)